File: LastSemesterInfo/AYDA Laboratorio/AYDA_Examen_1_student_distribution_(A2026)/linux/planning_a_company_party/c#/template.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Globalization;

class Employee {
    public double rating;
    public int left_child;
    public int right_sibling;
}

class Solution {
    public static double planParty(List<Employee> tree) {
        return 0.0;
    }

    static void Main(String[] args) {
        string firstLine = Console.ReadLine()!;
        if (string.IsNullOrEmpty(firstLine)) return;

        int n = int.Parse(firstLine.Trim());
        List<Employee> tree = new List<Employee>();

        for (int i = 0; i < n; ++i) {
            string[] tokens = Console.ReadLine()!.Trim().Split(' ');
            Employee emp = new Employee();
            emp.rating = double.Parse(tokens[0], CultureInfo.InvariantCulture);
            emp.left_child = int.Parse(tokens[1]);
            emp.right_sibling = int.Parse(tokens[2]);
            tree.Add(emp);
        }

        double result = planParty(tree);

        using (StreamWriter file = new StreamWriter("solution.txt")) {
            file.WriteLine(result.ToString("F2", CultureInfo.InvariantCulture));
        }
    }
}

File: LastSemesterInfo/AYDA Laboratorio/AYDA_Examen_1_student_distribution_(A2026)/linux/planning_a_company_party/rust/template.rs
use std::io::{self, Read};
use std::fs::File;
use std::io::Write;

struct Employee {
    rating: f64,
    left_child: i32,
    right_sibling: i32,
}

fn plan_party(tree: &[Employee]) -> f64 {
    0.0
}

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let mut tokens = input.split_whitespace();

    if let Some(n_str) = tokens.next() {
        let n: usize = n_str.parse().unwrap();
        let mut tree = Vec::with_capacity(n);

        for _ in 0..n {
            let rating: f64 = tokens.next().unwrap().parse().unwrap();
            let left_child: i32 = tokens.next().unwrap().parse().unwrap();
            let right_sibling: i32 = tokens.next().unwrap().parse().unwrap();
            tree.push(Employee { rating, left_child, right_sibling });
        }

        let result = plan_party(&tree);

        let mut file = File::create("solution.txt").unwrap();
        writeln!(file, "{:.2}", result).unwrap();
    }
}